### PR TITLE
feat #11 LIC9 and new coordinate angle function

### LINKED ISSUE #11: LIC9
There exists at least one set of three data points separated by exactly C PTS and D PTS
consecutive intervening points, respectively, that form an angle such that:
angle < (PI−EPSILON)
or
angle > (PI+EPSILON)
The second point of the set of three points is always the vertex of the angle. If either the first
point or the last point (or both) coincide with the vertex, the angle is undefined and the LIC
is not satisfied by those three points. When NUMPOINTS < 5, the condition is not met.
1 ≤ C PTS, 1 ≤ D PTS
C PTS+D PTS ≤ NUMPOINTS−3

### PR DESCRIPTION
**Pull request by agnesforsberg**
Closes #11

### CODE REVIEW

File: src/decide/Decider.java [hunk 1]
```diff
@@ -186,8 +222,44 @@ else if(a1>Math.PI/2 ||a2>Math.PI/2 ||a3>Math.PI/2){  // obtuse angle case cercl
...
+     consecutive intervening points, respectively, that form an angle such that:
+     angle < (PI−EPSILON)
+     or
+     angle > (PI+EPSILON)
+     The second point of the set of three points is always the vertex of the angle. If either the first
+     point or the last point (or both) coincide with the vertex, the angle is undefined and the LIC
+     is not satisfied by those three points. When NUMPOINTS < 5, the condition is not met.
+     1 ≤ C PTS, 1 ≤ D PTS
+     C PTS+D PTS ≤ NUMPOINTS−3
+
```
Comment (JustFredrik) [inline src/decide/Decider.java:235]: This method description seems misplaced between methods.

File: src/decide/Decider.java [hunk 2]
```diff
@@ -186,8 +222,44 @@ else if(a1>Math.PI/2 ||a2>Math.PI/2 ||a3>Math.PI/2){  // obtuse angle case cercl
...
+     @return  Boolean representing if the condition is met or not**/
     public boolean lic9(){
-        return true;
+        // False if preconditions not met
```
Comment (JustFredrik) [inline src/decide/Decider.java:238]: missing method description.

File: src/decide/Decider.java [hunk 3]
```diff
@@ -93,7 +95,41 @@ public boolean lic3(){
     }
 
     public boolean lic4(){
-        return true;
+        // false in case of invalid parameters
+        if(this.parameters.Q_PTS < 2 || this.parameters.Q_PTS > numpoints){
+            return false;
```
Comment (JustFredrik) [inline src/decide/Decider.java:100]: Missing method description.

**Review by JustFredrik — CHANGES_REQUESTED** [review]

File: src/decide/Decider.java (hunk 3 above)
Comment (agnesforsberg) [reply src/decide/Decider.java:100]: It's added in the commit for that issue, this was just branched from that branch so not completely accurate.

**Pull request merged**